### PR TITLE
Team sigchain stricter checks on owner invites

### PR DESCRIPTION
**Pull request by mlsteele**
Non-owners can't invite owners. If this is wrong or disagrees with the real wild world, then teams will be unloadable. Better block these shenanigans on the server and scan existing teams. I don't want to merge this until those are done.

### CODE REVIEW

**Comment by zapu** [comment]
So the current state is that no one can be invited as an owner, no matter by whom, unless it's an implicit team. Is this necessarily a bad thing? We shouldn't have many PUKless cases now, unless we consider in-between resets, in which the admins should really not invite users like that at all.

i'm open for being convinced otherwise though!

**Comment by zapu** [comment]
We are also disallowing owner social invites on RPC level: https://github.com/keybase/client/blob/master/go/teams/teams.go#L897-L899

But I guess with this PR we are positioning ourselves to support admin->owner promotion. But I don't see why we should allow "inviting" owners.

**Comment by mlsteele** [comment]
> So the current state is that no one can be invited as an owner, no matter by whom, unless it's an implicit team.

This isn't what I've observed. This may be made more confusing by the difference in terminology of invite/add in the realms of sigchain and user experience. An owner can sigchain-invite another owner if they have no puk (website user).

**Comment by mlsteele** [comment]
This PR clamps down on admins trying to sigchain-invite owners. Which seems like it ought to be disallowed. I observed this happening locally and it roundtripped.

**Comment by mlsteele** [comment]
Scanned prod teams and they're all ok with this change. Given the exceptions added to chain.go. With @pzduniak's server change no new links should cross the line. 

@maxtaco the sig ID comes from `DecodeOuterLinkV2` so should be good.

@zapu what do you think?

File: go/teams/chain.go [hunk 1]
```diff
@@ -1719,7 +1737,15 @@ func (t *teamSigchainPlayer) sanityCheckInvites(
 	if invites.Owners != nil && len(*invites.Owners) > 0 {
 		additions[keybase1.TeamRole_OWNER] = nil
 		for _, i := range *invites.Owners {
-			if !options.implicitTeam && !assertIsKeybaseInvite(t.G(), i) {
+			if !options.isRootTeam {
+				return nil, nil, fmt.Errorf("encountered invite of owner in non-root team")
+			}
+			if !signerIsExplicitOwner {
+				if !hardcodedInviteRuleExceptionSigIDs[sigID] {
+					return nil, nil, fmt.Errorf("encountered invite of owner by non-owner")
```
Comment (zapu) [inline go/teams/chain.go]: I think a good test would be to try to post a link like that (bypassing service_helper functions and transactions code) and having it caught by precheck. There are some tests like this for implicit teams if you grep for `PrecheckAppendError`.

**Pull request merged**